jetty-util/src/main/java/org/eclipse/jetty/util/PathUtil.java

```diff
@@ -0,0 +1,157 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2021 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.util;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+
+/**
+ * Collection of Path encoding / decoding utilities for URI, Http Headers, and HTML
+ */
+public class PathUtil
```

Comment:
I don't think this is well named.   It reads like it is for utility methods for the `Path` class.

Instead it has methods that work on the strange decoded URI path space that is ill defined by the servlet specification.    Currently we have such methods on URIUtil, which I guess is similarly badly named as they are not URI paths either.... well I guess they are decoded URI paths, so it's a little bit better.

Eitherway, I think we should strive to not add extra classes like this, unless their purpose is clearly defined.